src/buildMathML.js

```diff
@@ -23,6 +23,16 @@ var makeText = function(text, mode) {
     return new mathMLTree.TextNode(text);
 };
 
+// Keep track of the current font specified by /mathbb, /mathit, or /mathrm
+var fontStack = [];
```

Comment:
Unfortunately, we can't use the same approach used for color b/c the mathvariant attribute is only valid on <mi>, <mn>, <mo>, <ms>, <mtext> so we have to set it on each of them individually.  This means we have to keep track of the current font.